android/src/main/java/com/instabug/reactlibrary/RNInstabugCrashReportingModule.java

```diff
@@ -79,41 +85,61 @@ public void run() {
      * Send handled JS error object
      *
      * @param exceptionObject Exception object to be sent to Instabug's servers
+     * @param userAttributes  (Optional) extra user attributes attached to the crash
+     * @param fingerprint     (Optional) key used to customize how crashes are grouped together
+     * @param levelType       different severity levels for errors
      */
     @ReactMethod
-    public void sendHandledJSCrash(final String exceptionObject) {
+    public void sendHandledJSCrash(final String exceptionObject, @Nullable ReadableMap userAttributes, @Nullable String fingerprint, @Nullable String levelType) {
         try {
             JSONObject jsonObject = new JSONObject(exceptionObject);
-            sendJSCrashByReflection(jsonObject, true, null);
-        } catch (Exception e) {
+            MainThreadHandler.runOnMainThread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Method method = getMethod(Class.forName("com.instabug.crash.CrashReporting"), "reportException", JSONObject.class, boolean.class,
+                                Map.class, JSONObject.class, IBGNonFatalException.Level.class);
+                        if (method != null) {
+                            IBGNonFatalException.Level nonFatalExceptionLevelType = ArgsRegistry.nonFatalExceptionLevel.getOrDefault(levelType, IBGNonFatalException.Level.ERROR);
+                            method.invoke(null, jsonObject, true, userAttributes == null ? null : userAttributes.toHashMap(), fingerprint == null ? null : CrashReporting.getFingerprintObject(fingerprint), nonFatalExceptionLevelType);
```

Comment:
I feel like it's a bit cluttered to have all these ternary statements in the arguments, why don't we extract them into variables instead and then just pass the variables to `method.invoke`?